app/src/main/kotlin/com/wire/android/ui/home/newconversation/NewConversationViewModel.kt

```diff
@@ -111,15 +112,18 @@ class NewConversationViewModel
     }
 
     private suspend fun searchKnown(searchTerm: String) {
-        // TODO: this is going to be refactored on the Kalium side so that we do not use Flow
-        searchKnownUsers(searchTerm).onStart {
-            localContactSearchResult = ContactSearchResult.InternalContact(SearchResultState.InProgress)
-        }.catch {
-            localContactSearchResult = ContactSearchResult.InternalContact(SearchResultState.Failure())
-        }.flowOn(Dispatchers.IO).collect {
-            localContactSearchResult = ContactSearchResult.InternalContact(
-                SearchResultState.Success(it.result.map { otherUser -> otherUser.toContact() })
-            )
+        localContactSearchResult = ContactSearchResult.InternalContact(SearchResultState.InProgress)
+
+        viewModelScope.launch {
+            localContactSearchResult = try {
+                val searchResult = searchKnownUsers(searchTerm)
+
+                ContactSearchResult.InternalContact(
+                    SearchResultState.Success(searchResult.result.map { otherUser -> otherUser.toContact() })
+                )
+            } catch (exception: Exception) {
```

Comment:
I think detekt will complain about having richer type of exceptions, so we should try to be more concrete or add a `Suppress` header for the method